src/Geno/C++/GUI/MainWindow.cpp

```diff
@@ -427,14 +434,22 @@ void* MainWindow::ImGuiSettingsReadOpenCB( ImGuiContext* /*pContext*/, ImGuiSett
 
 void MainWindow::ImGuiSettingsReadLineCB( ImGuiContext* /*pContext*/, ImGuiSettingsHandler* pHandler, void* pEntry, const char* pLine )
 {
-	MainWindow* pSelf = ( MainWindow* )pHandler->UserData;
-	const char* pName = ( const char* )pEntry;
-	int         Bool;
+	MainWindow*           pSelf = ( MainWindow* )pHandler->UserData;
+	const char*           pName = ( const char* )pEntry;
+	int                   Bool;
+	char                  Path[ 1024 ];
 
 	if(      strcmp( pName, "Text Edit" ) == 0 ) { if( sscanf( pLine, "Active=%d", &Bool ) == 1 ) pSelf->pTitleBar->ShowTextEdit          = Bool; }
 	else if( strcmp( pName, "Workspace" ) == 0 ) { if( sscanf( pLine, "Active=%d", &Bool ) == 1 ) pSelf->pTitleBar->ShowWorkspaceOutliner = Bool; }
 	else if( strcmp( pName, "Output"    ) == 0 ) { if( sscanf( pLine, "Active=%d", &Bool ) == 1 ) pSelf->pTitleBar->ShowOutputWindow      = Bool; }
 
+	// Recent Workspaces
+#if defined( _WIN32 )
+	if( sscanf( pLine, "Path=%s", Path, ( unsigned )sizeof( Path ) ) == 1 ) pSelf->AddRecentWorkspace( Path );
+#elif defined( __linux__ ) || defined( __APPLE__ ) // _WIN32
+	if( sscanf( pLine, "Path=%s", Path ) == 1 ) pSelf->AddRecentWorkspace( Path );
+#endif //  __linux__ || __APPLE__ 
+
```

Comment:
sscanf will stop scanning when it reaches whitespace, which means that it's going to mess up paths with spaces in them!

Reply:
What we could do is check if the line starts with "Path=" and then just shove everything after it into AddRecentWorkspace